### PR TITLE
Add Correct DAS Documentation

### PR DESCRIPTION
**Pull request by Blake-Haydon**
## Description

This PR addresses a major pain point of using any of the DAS command line tools, the fact that the commands are confusing. 

A few fun things
- Renamed logger
- Beefed up docs with solid example
- Hyperlinking relative docs (means they won't break)

## Screenshots

image

## Steps to Test

Just have a look at the new documentation [here](https://github.com/monash-human-power/data-acquisition-system/blob/blake-haydon/clean-out-old-mqtt-recorder/DAS/README.md)

Don't stress about anything else apart from the readme. The other changes have all been merged it just looks like a lot as I branched off my other work.

### CODE REVIEW

**Comment by emilytrau** [comment]
I think this needs to be rebased from master

**Comment by hallgchris** [comment]
Would you mind merging/rebasing master? Will need to be done anyway before this PR can be merged in but it'll be much easier to review :)

**Comment by Blake-Haydon** [comment]
Just merged before, should be able to merge into master now!

File: DAS/README.md
Comment (hallgchris) [inline DAS/README.md]: Out of interest, what's the reasoning behind using `python -m das.mqtt_recorder` over `python das/mqtt_recorder.py`?

File: DAS/README.md
Comment (hallgchris) [inline DAS/README.md]: Since this is a documentation PR, I'll be pedantic - add a full stop

File: DAS/README.md
Comment (hallgchris) [inline DAS/README.md]: Maybe consider making a column for the default value?

Would be awesome if we could generate this automatically from the `argparse` usage... if only

File: DAS/das/utils/mqtt_test.py
Comment (hallgchris) [inline DAS/das/utils/mqtt_test.py]: What makes this a `util` and `mqtt_playback`/`fake_module` not, despite them being effectively the same thing, just for different bike versions? I feel like this could be in the directory above.

Also while we're moving it, we should rename it too. I'm tempted to suggest `mqtt_playback_v2.py`... Although that could clearly lead to some confusion due to the original name of the second iteration of your logger. Open to suggestions, would be pretty easy to beat `mqtt_test.py`

File: DAS/README.md
Comment (hallgchris) [inline DAS/README.md]: Another assumption is that we have poetry installed.

File: README.md
Comment (hallgchris) [inline README.md:11]: Just noticed that the url above is outdated (although it does still work). Could you update it quickly?

File: README.md
Comment (hallgchris) [inline README.md]: Probably keep the `.`

**Review by hallgchris — CHANGES_REQUESTED** [review]
Sorry for the nitpicks, just like some full stops and stuff.

Really love the relative links in the readmes, I didn't know we could do that!

File: DAS/README.md
Comment (Blake-Haydon) [reply DAS/README.md]: It is because the folder structure is a python module so `-m` is needed to load everything in correctly.

File: DAS/README.md
Comment (Blake-Haydon) [reply DAS/README.md]: Good idea!

File: DAS/das/utils/mqtt_test.py
Comment (Blake-Haydon) [reply DAS/das/utils/mqtt_test.py]: I think that is a very good idea actually, I didn't realise that `mqtt_test` was essentially a mqtt playback tool. It definitely should not be in the `util` folder.

I think the easiest would be to name all of the tools that I wrote as "V3" and rename `mqtt_test.py` to `mqtt_playback_v2.py`

Comment (Blake-Haydon) [reply DAS/das/utils/mqtt_test.py]: After some thought I have decided to prepend all files corresponding to a version of the logger/recorder with either "V3" or "V2". This also makes it really easy to search tools now.

**Comment by Blake-Haydon** [comment]
All of the requested changes have been fixed!
Check out the new [DAS README](https://github.com/monash-human-power/data-acquisition-system/blob/blake-haydon/clean-out-old-mqtt-recorder/DAS/README.md) and [Repo README](https://github.com/monash-human-power/data-acquisition-system/blob/blake-haydon/clean-out-old-mqtt-recorder/README.md) to see the changes proposed in this PR.

File: DAS/README.md
Comment (hallgchris) [reply DAS/README.md]: Still not completely sure on this, for me everything seems to load properly either way.

**Review by hallgchris — DISMISSED** [review]
This documentation is incredible! Looks great to me now.

Btw, going to swap Khangus for a secondary reviewer to take a look.

File: DAS/das/V2_mqtt_playback.py
Comment (kdav108) [inline DAS/das/V2_mqtt_playback.py]: I think @hallgchris has a PR to add these topics to the common repo, so maybe we should wait for that PR to be merged in and then change these to use the topics directly from `mhp.topics`. It may be annoying to have another PR to just change this later on.

File: DAS/README.md
Comment (kdav108) [inline DAS/README.md]: May help putting the units for the time length here.

**Review by kdav108 — COMMENTED** [review]
Great documentation! The links to the python files is quite cool. We should get Blake to work on documentation on other repositories as well 😂. 
I just have a few suggestions, other than that this looks good!

File: DAS/das/V2_mqtt_playback.py
Comment (hallgchris) [reply DAS/das/V2_mqtt_playback.py]: Nah I reckon it's fine. Whichever PR is ready second updates any changes from master. It shouldn't be too great of a change to make!

Comment (Blake-Haydon) [reply DAS/das/V2_mqtt_playback.py]: I think that it should be easy to fix in another PR.

**Comment by Blake-Haydon** [comment]
The units have been added to the documentation @kdav108.

**Review by kdav108 — DISMISSED** [review]

**Review by hallgchris — DISMISSED** [review]
LGTM!

File: DAS/README.md
Comment (kdav108) [inline DAS/README.md]: I know you didn't write it, but could you remove the duplicate `data` here?

File: DAS/README.md
Comment (kdav108) [inline DAS/README.md]: The topics have been updated in #66, so just confirming whether we'll be changing these to the updated topics in a separate PR or shall we do it here (if it's a separate PR, we should prioritise that so we don't have inconsistent documentations lying around)

**Review by kdav108 — CHANGES_REQUESTED** [review]
Almost done! Just some small (pedantic) stuff :)

File: DAS/README.md
Comment (hallgchris) [reply DAS/README.md]: Since #66 has been merged, this PR should have its topics updated. No big deal, a very quick fix!

**Review by hallgchris — CHANGES_REQUESTED** [review]

File: DAS/README.md
Comment (Blake-Haydon) [reply DAS/README.md]: Too easy, thanks!

**Comment by Blake-Haydon** [comment]
This has taken forever but I think it is finally ready to be merged. Incorrect topic names and duplicate words have been addressed!

**Review by kdav108 — DISMISSED** [review]
Not sure if @hallgchris meant that we should update all old mqtt topics used in this PR (like on line 78 of `V2_mqtt_playback.py`) and import them from `mhp` or just update it in the `README.md` where he had commented. Other than that this looks good to me!

File: DAS/das/V2_mqtt_playback.py
Comment (hallgchris) [inline DAS/das/V2_mqtt_playback.py:1]: Sorry, just realised another thing - this is meant to replace `mqtt_test.py` isn't it? Right now it only duplicates it.
Remove `mqtt_test.py` and copy over the new `mhp` topic usages into here. Then good to go, promise!

Sorry for taking forever btw. Will be quicker this time.

**Review by hallgchris — CHANGES_REQUESTED** [review]

File: DAS/das/V2_mqtt_playback.py
Comment (Blake-Haydon) [reply DAS/das/V2_mqtt_playback.py:1]: Should be fixed now!

**Comment by Blake-Haydon** [comment]
@hallgchris The duplicate file and the mhp topic usage should be fixed now.

**Review by hallgchris — APPROVED** [review]
Looks good! Sorry for the lengthy review haha

**Pull request merged**
